docs/reference.rst

```diff
@@ -298,6 +298,17 @@ For more information about using the streaming support of RAUC, refer to
   This option can be used to set the path of the CA certificate which should be
   used instead of the system wide store of trusted TLS/HTTPS certificates.
 
+``send-headers``
+  This option takes a list of information to send as HTTP header fields.
+  Supported values are:
+
+  * ``boot-id``: Enables sending the *boot_id* as ``X-RAUC-BOOT-ID`` header field.
+  * ``machine-id``: Enables sending the *machine-id* as ``X-RAUC-MACHINE-ID`` header field.
```

Comment:
jfyi, [Man page says: ](https://www.freedesktop.org/software/systemd/man/machine-id.html):
> This ID uniquely identifies the host. It should be considered "confidential", and must not be exposed in untrusted environments, in particular on the network.

I am not sure if that warrants using `sd_id128_get_machine_app_specific` or if we can just expect the user not to download over HTTP if they care about this.

Reply:
We should point to the documentation and mention that this header should not be used over unauthenticated connections or if the server is not trusted. In most cases, I expect that the update server needs to know the real `machine-id` tough. If needed, we could add `*-app-specific` variants to `boot-id` and `machine-id` later.

Reply:
Will add a note for this option.